### PR TITLE
Change drop_outlier default to 500

### LINKED ISSUE #237: Questions about drop_outlier_features
I was looking at some code @fefossa is writing, and came across `drop_outlier_features`, which was added in #62 but somehow escaped my notice (it isn't on in the profiling template, for example). This is potentially super useful to us, but I was wondering about the default value (15). I know I'm asking about decisions made in December 2019, a far simpler time, but do you recall how this value was chosen? It doesn't appear at first pass to be documented, and this seems to me to be WAY too low - we routinely see features we believe in above that absolute value. Fernanda is going to check on some profiles with and without that feature reduction step added to get a sense of in her data how much effect it is having, but it is possibly significant.

I know the parameter is configurable, so we can override it if we'd like, but in general I assume we want defaults to be broadly applicable for the vast set of use cases, and I don't know that this one is.

### PR DESCRIPTION
**Pull request by bethac07**
_modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_

# Description
Resolves #237 by changing the drop_outliers default cutoff from 15 to 500, for reasons more thoroughly detailed in that issue. 

## What is the nature of your change?

- [x] Bug fix (fixes an issue).

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: pycytominer/cyto_utils/features.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def drop_outlier_features(
         Features present in the population dataframe. If "infer", then assume Cell Painting features are those that start with "Cells_", "Nuclei_", or "Cytoplasm_"
     samples : list of str or str, default "all"
         Samples to perform the operation on
-    outlier_cutoff : int or float, default 15
+    outlier_cutoff : int or float, default 500
```
Comment (gwaybio) [inline pycytominer/cyto_utils/features.py:157]:
```suggestion
    outlier_cutoff : int or float, default 500
    see https://github.com/cytomining/pycytominer/issues/237 for details.
```

Adding this explicit pointer for potential future reference

File: pycytominer/tests/test_cyto_utils/test_feature_drop_outlier.py [hunk 2]
```diff
@@ -23,29 +23,34 @@
...
 
 def test_outlier_default():
     result = drop_outlier_features(data_df)
-    expected_result = ["Cells_zz", "Nuclei_z"]
+    expected_result = ["Cells_zz"]
     assert sorted(result) == sorted(expected_result)
 
 
-def test_outlier_high_cutoff():
+def test_outlier_30_cutoff():
```
Comment (gwaybio) [inline pycytominer/tests/test_cyto_utils/test_feature_drop_outlier.py:37]: This test and `test_outlier_default()` seem essentially the same to me. We can probably delete this one, especially given the addition of `test_outlier_15_cutoff()`

**Review by gwaybio — APPROVED** [review]
LGTM, a couple suggestions.

@niranjchandrasekaran - would you like to take a look?

**Comment by bethac07** [comment]
@gwaybio Addressed your comments, once the tests rerun should we wait for Niranj to check or just pull?

**Pull request merged**